tests/fixtures_file_writer.py

```diff
@@ -347,3 +350,89 @@ def fixture_running_four_board_file_writer_process(runnable_four_board_file_writ
 
     fw_process.stop()
     fw_process.join()
+
+
+def file_writer_process_with_closed_h5_files_for_upload(
```

Comment:
This name reads like a fixture but it is just a function that uses a `four_board_file_writer_process`. Could probably change it to something like `create_closed_h5_files_for_upload`. Something else to watch out for is that you actually run the process in this function. There is a `invoke_process_run_and_check_errors(file_writer_process, num_iterations=8)` in a test that uses this function and it's unclear at a glance what the number 8 does. If those iterations were done here then there would be no need for the user of this function to figure out how many loops to run file_writer. Also be sure to use the minimum number of iterations necessary (typically one iteration per item in the queue, can be more if that item causes something to happen on the next iteration though), and if applicable drain any queues to make assertions on them easier in tests